### PR TITLE
Implement Module.{bindings, dependencies} 

### LINKED ISSUE #329: Implement Module.{bindings, dependencies}
Split from #320

https://github.com/morganey-lang/Morganey/blob/458946d8fefc71fffc8c98834a7752c9081925ab/src/main/scala/me/rexim/morganey/module/Module.scala#L10

### PR DESCRIPTION
**Pull request by rexim**
Close #329

### CODE REVIEW

File: src/main/scala/me/rexim/morganey/module/Module.scala [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+import java.net.URL
+
 class Module(modulePath: ModulePath, classLoader: ClassLoader = Module.getClass.getClassLoader) {
   def canonicalPath: String =
     modulePath.asCanonicalPath.path
 
-  // TODO: Implement Module.{bindings, dependencies}
-  def bindings: Try[Set[MorganeyBinding]] = ???
-  def dependencies: Try[Set[Module]] = ???
+  private def nodes: Try[List[MorganeyNode]] = {
```
Comment (ForNeVeR) [inline src/main/scala/me/rexim/morganey/module/Module.scala:14]: Is it possible to make it `lazy val nodes`?

**Review by ForNeVeR — APPROVED** [review]

File: src/main/scala/me/rexim/morganey/module/Module.scala (hunk 1 above)
Comment (rexim) [reply src/main/scala/me/rexim/morganey/module/Module.scala:14]: Yeah, I think it is, but my trail of thoughts was the following: `lazy val` essentially caches the module content. If we change it the module object will still return the "cached" values and to reload the module we have to create the same instance again.

Comment (ForNeVeR) [reply src/main/scala/me/rexim/morganey/module/Module.scala:14]: Alright, let's leave it as-is.

**Review by keddelzz — APPROVED** [review]

**Comment by rexim** [comment]
@ForNeVeR @keddelzz thank you so much guys for the review!

**Pull request merged**